pkg/lifecycle/lifecycle.go

```diff
@@ -19,22 +19,40 @@ import (
 var log = zap.New(zap.UseDevMode(true)).WithName(color.Print("lifecycle", color.Green))
 
 func GetPodFromDaemonSet(key types.NamespacedName) unstructured.UnstructuredList {
+	ds := &unstructured.Unstructured{}
+	ds.SetAPIVersion("apps/v1")
+	ds.SetKind("DaemonSet")
 
-	pl := unstructured.UnstructuredList{}
-	pl.SetKind("PodList")
-	pl.SetAPIVersion("v1")
+	err := clients.Interface.Get(context.TODO(), key, ds)
+	if apierrors.IsNotFound(err) || err != nil {
+		warn.OnError(err)
+		return unstructured.UnstructuredList{}
+	}
+
+	return getPodListForUpperObject(ds)
+}
+
+func GetPodFromDeployment(key types.NamespacedName) unstructured.UnstructuredList {
 
 	ds := &unstructured.Unstructured{}
```

Comment:
I think the `ds` variable name is no longer appropriate for a Deployment.
Also, could we move to typed objects instead of `*unstructured.Unstructured`? For such a change, we might want to wait for unit tests to be there, though.

Reply:
fixing ds variable.
Regarding typed objects: i think it is a more global change, since currently SRO code uses unstructured almost exclusively. 
We can open a task on it and check if this is feasible: there are a lot of places, (e.g poll.go) where using unstructured gives you an ability to use the same parameter in different function

Reply:
fixed the "ds"